src/core/common/message.hpp

```diff
@@ -87,10 +87,11 @@ struct MessageInfo
         PriorityQueue *mPriority; ///< Identifies the priority queue (if any) where this message is queued.
     } mQueue;                     ///< Identifies the queue (if any) where this message is queued.
 
-    uint16_t    mReserved;    ///< Number of header bytes reserved for the message.
-    uint16_t    mLength;      ///< Number of bytes within the message.
-    uint16_t    mOffset;      ///< A byte offset within the message.
-    uint16_t    mDatagramTag; ///< The datagram tag used for 6LoWPAN fragmentation.
+    uint16_t mReserved;       ///< Number of header bytes reserved for the message.
+    uint16_t mLength;         ///< Number of bytes within the message.
+    uint16_t mOffset;         ///< A byte offset within the message.
+    uint32_t mDatagramTag;    ///< The datagram tag used for 6LoWPAN fragmentation or identification used for IPv6
```

Comment:
nit: move above `mReserved` to  reduce likelihood of padding.

Reply:
Moved.